### DISCUSSION TITLE
Netzladung per Trigger möglich ?

### DISCUSSION THREAD

**Discussion by dancekid** [spali/go-rscp#9] [Q&A] [answered]
Guten Abend Spali,

gibt es einen Tag zum aktiven Netzladen den ich triggern kann ?

In der S10 Gui kann man einmal am Tag eine manuelle Netzladung von bis zu 3 kWh starten.
Das würde ich gerne automatisch mit ioBroker initiieren.

Vorhandener NodeRed Flow:
Die preisgünstigen Stunden von Awattar.de ermittelt der ioBroker und setzt SG Ready Eingang 1 (heizen) = EIN (1) und bei teuren Stunden  SG Ready Eingang 2 (vermeiden)= EIN (1).

Gruß Michael

**Comment by spali** [comment, marked as answer]
Hallo Michael

Es gibt [PVI_REQ_SYSTEM_MODE](https://github.com/spali/go-rscp/blob/28a0ed5d8e1b4a347d9c45e937f54f3b9bcf1ec7/rscp/tag.go#L680). Wobei das spätestens alle 30 Sekunden neu gesetzt werden muss. Ich konnte den Speicher damit mal dazu bringen "forciert" zu laden. Jedoch blicke ich da auch nicht 100% durch was die anderen Modi sind und was die Nebeneffekte sind. Darum habe ich die Finger davon gelassen.

Die angesprochene manuelle Netzladung kannst du mit [EMS_REQ_START_MANUAL_CHARGE](https://github.com/spali/go-rscp/blob/28a0ed5d8e1b4a347d9c45e937f54f3b9bcf1ec7/rscp/tag.go#L282) starten.
In meinem Fall, lade ich den Speicher Nachts damit wiederholt, wenn erwartet wird, dass kein Strom vom Dach kommt am Folgetag.
Es geht in unserem Fall sogar mehrmals hintereinander, wobei nach einer Ladung immer für ca. 1 Stunde gesperrt ist. Dann kann ich die manuelle Ladung jedoch erneut anstossen. Mit ein wenig Logik kann ich so den Speicher über Nacht auf einen gewünschten Ladepegel bringen.

**Reply by dancekid** [reply]
Hallo Spali,
wie macht man denn aus den Dateien eine .exe für die Nutzung in Windows ?

**Reply by spali** [reply]
Musst du nicht selber machen: https://github.com/spali/go-rscp/releases/tag/v0.1.0, dort einfach z.B. e3dc_0.1.0_Windows_x86_64.zip für ein 64bit Windows herunteladen.

**Reply by dancekid** [reply]
danke für die Info.

Kann man per RSCP auch die Notstromreserve ändern ?
Würde gerne an einem Sonnentag die Reserve ab 0:00 von 60% auf 30% runtersetzen.
Bei Sonne vor einem Regentag von 30% auf 60% hochsetzen.
100% sind bei mir 17500 Wh. Entsprechend 30% und 10.000 und 5.000 sollte ich dann übergeben.

**Reply by spali** [reply]
Ich vermute ja, kann es dir aber nicht sagen. Hab mich mit der Notstromreserve nicht so weit beschäftigt, da die bei meiner Quattroporte nicht den gleichen Funktions-Umfang hat wie eine S10.
Du müsstest dich durch die Entwickler Doku deiner S10 wühlen und hoffen, dass die Tags gut genug dokumentiert sind um da was hin zu bekommen. 
Leider ist RSCP von E3DC nicht so gut dokumentiert, dass es selbsterklärend ist... man muss sehr viel ausprobieren und tüfteln. 
Dies, und dass alle Modelle unterschiedlich sind, ist auch der Grund, wieso ich hier nur eine Library mit sehr tiefem Level gebaut habe ohne Spezifika der Implementierung.

**Reply by dancekid** [reply]
kannst du mir bitte noch den Syntax sagen für Entladeleistung ? Möchte die derzeitig feste Entladesperre von 0.00 - 06.00 umstellen auf dynamisch und es pro Stunde ändern
EMS_GET_POWER_SETTINGS mit Tag MAX_CHARGE_POWER und einmal 1 und einmal 3000 ? Wie ist die Syntax ?

**Reply by spali** [reply]
Meinst du nun Entladesperre oder Ladesperre (`IDLE_PERIODS` bei Tags)... bzw. Reden wir von der Sperre oder der Leistungslimitierung.
`MAX_CHARGE_POWER` wäre die Ladeleistungslimitierung... es gibt da auch noch die Entladeleistunglimitierung.
Falls du die Sperren (`IDLE_PERIODS`) setzen willst... dann wird der Syntax schon einiges komplexer.
Für die Limitierung `MAX_...`  musst du diese mit dem `SET` Tag setzen... es gibt meistens zu jeder Einstellung ein `GET` und ein `SET`.
Und für die `SET` tags welche Container sind, musst du dann nur die Parameter als Array darin "schachteln".
Schnapp dir das Excel mit den Tags von E3DC und tüftle ein wenig. Dann solltest du es hinbekommen.
Wenn ich mich recht erinnere, sind für `SET_POWER_SETTINGS` nur die Unterparameter nötigt die du auch verändern willst. Musst also nicht zwingend jedes mal alles senden, wenn es sich nicht verändert hat.

**Reply by valbuz** [reply]
> danke für die Info.
> 
> Kann man per RSCP auch die Notstromreserve ändern ? Würde gerne an einem Sonnentag die Reserve ab 0:00 von 60% auf 30% runtersetzen. Bei Sonne vor einem Regentag von 30% auf 60% hochsetzen. 100% sind bei mir 17500 Wh. Entsprechend 30% und 10.000 und 5.000 sollte ich dann übergeben.

Hast Du da ne Lösung gefunden?
Danke
